File: frontend/src/types/category.ts
export type ExpenseType = 'monthly' | 'annual';

export interface Category {
  id: string;
  name: string;
  icon: string;
  color: string;
  expenseType: ExpenseType;
  displayOrder?: number;
  subCategories?: SubCategory[];
}

export interface SubCategory {
  id: string;
  name: string;
  categoryId: string;
  icon?: string;
  budgetLimit?: number;
  displayOrder?: number;
  fixedAmount?: number;
  isMandatory?: boolean;
}

export interface CreateCategoryDTO {
  name: string;
  icon: string;
  color: string;
  expenseType: ExpenseType;
  displayOrder?: number;
}

export interface CreateSubCategoryDTO {
  name: string;
  icon?: string;
  budgetLimit?: number;
  displayOrder?: number;
  fixedAmount?: number;
  isMandatory?: boolean;
}

File: backend/src/main/java/com/houseexpenses/model/Category.java
package com.houseexpenses.model;

import jakarta.persistence.*;
import lombok.*;
import java.time.OffsetDateTime;
import java.util.ArrayList;
import java.util.List;
import java.util.UUID;

@Entity
@Table(name = "categories")
@Data
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class Category {

    @Id
    @GeneratedValue(strategy = GenerationType.UUID)
    private UUID id;

    @Column(name = "user_id", nullable = false)
    private UUID userId;

    @Column(nullable = false, length = 100)
    private String name;

    @Column(nullable = false, length = 50)
    private String icon;

    @Column(nullable = false, length = 7)
    private String color;

    @Column(name = "expense_type", nullable = false, length = 10)
    @Enumerated(EnumType.STRING)
    private ExpenseType expenseType;

    @Column(name = "display_order")
    @Builder.Default
    private Integer displayOrder = 0;

    @Column(name = "is_active")
    @Builder.Default
    private Boolean isActive = true;

    @Column(name = "created_at")
    private OffsetDateTime createdAt;

    @Column(name = "updated_at")
    private OffsetDateTime updatedAt;

    @OneToMany(mappedBy = "category", cascade = CascadeType.ALL, orphanRemoval = true)
    @Builder.Default
    private List<SubCategory> subCategories = new ArrayList<>();

    @PrePersist
    protected void onCreate() {
        createdAt = OffsetDateTime.now();
        updatedAt = OffsetDateTime.now();
    }

    @PreUpdate
    protected void onUpdate() {
        updatedAt = OffsetDateTime.now();
    }

    public enum ExpenseType {
        monthly, annual
    }
}
